game-app/game-core/src/main/java/tools/image/FileHelper.java

```diff
@@ -0,0 +1,21 @@
+package tools.image;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class FileHelper {
+
+  public static Path getFileInMapRoot(
+      final Path mapFolderLocation, final String mapName, final String fileName) {
+    Path path = null;
+    if (mapFolderLocation != null && Files.exists(mapFolderLocation)) {
+      path = mapFolderLocation.resolve(fileName);
+    }
+    if (path == null || !Files.exists(path)) {
+      path = Path.of(mapName).resolveSibling(fileName);
```

Comment:
Critical:  A 'mapName' can now be completely arbitrary, it may or may not at all match the map folder or any of the folders within a map. A map name is an arbitrary string in 'map.yml', it now no longer has any required relationship to map repo name or the folder name.

Given that, is this going off of the legacy design where the map name and map folder used to match?

Reply:
Please have a look at the latest changes. It turns out that I could turn some Strings into Path objects and it makes a lot more sense now :)

Reply:
We might not be on the same page. My point is the statement `Path.of(mapName)` is wrong. The `mapName` is _not_ the name of the map folder, the two are no longer the same. The map name is now decoupled from the file system. Hence, deriving the map name from map folder is no longer correct.

Reply:
@DanVanAtta I got that, however I think the at least this part of the code doesn't really care about the name of the map, I realized that this "mapName" that was supplied to `Path#of` was in fact the path to the map converted into a string at some point.
So yeah, you're right that mapName is misleading and I tried to change that to mapFolder whereever I saw it, but I think the latest code fixes some parts of this misleading structure

Reply:
Let's see if we can avoid digging a hole deeper and maybe rename variables if it's only a semantic difference now. I'd suggest we make the class more specific as well, eg: `MapFileFinder`, and then see if we can avoid the 'mapName' name and call it something else. Perhaps ideally see if we can pass in the root folder first. There is actually already a util in `FileUtils` to do exactly that: `FileUtils#find`

Reply:
Renaming the class sounds reasonable, is there anything else that should be done as part of this PR?
`FileUtils#find` is similar but not quite the same, I don't quite see how it can be used to shorten this helper method.

'mapName' isn't in the code anymore AFAIK, so there isn't anything to 'correct' there

Reply:
Thanks, for the latter point, I am confused now.

I see the following:
```
  path = Path.of(mapName).resolveSibling(fileName);
```
Where `mapName` is a parameter. The above is calling something a mapName, which used to be completely synonmous with the folder name, and using that as a path parameter. Is this just as simple as renaming that as `mapFolderPath` or something like that?

Reply:
The code you're seeing is outdated, have a look at the full diff and make sure you're looking at all commits